packages/ember-data/lib/system/store/factory-locators.js

```diff
@@ -0,0 +1,74 @@
+import normalizeModelName from 'ember-data/system/normalize-model-name';
+
+function modelFor(container, modelName) {
+  var factory = modelFactoryFor(container, modelName);
+
+  if (!factory) {
+    //Support looking up mixins as base types for polymorphic relationships
+    factory = modelForMixin(container, modelName);
+  }
+  if (!factory) {
+    throw new Ember.Error("No model was found for '" + modelName + "'");
+  }
+  factory.modelName = factory.modelName || normalizeModelName(modelName);
+  factory.__container__ = factory.__container__ || container;
```

Comment:
in slack we discussed this further. I provided the following example:

As nothing prevents two apps from having two different set of resolvers, factory level reflection without polluting instance state isn't really available to us.

The solution is to continue to purge instance state (the container/ store etc) from factories, and instead enable the needing wiring when instance information is present.